torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -73,6 +76,11 @@ enum ErrorHandlingMode {
 
 #define SHOULD_TEAR_DOWN(a) (a != NoHandling && a != CleanUpOnly)
 
+#define PRINT_COLLECTIVE_HASH_SIGNATURE(rank, opType, numel, hashValue) \
+  LOG(ERROR) << "[RANK" << rank << "] : Collective hash of " << opType  \
```

Comment:
I used ERROR just to avoid any possible message missing issues. But it seems that WARN is enough. This is not critical issue though since you also use a gating variable.

Reply:
Not sure if we can also add the sequence number. I remember PG has a sequence number.

Reply:
yes, but this is not available in python..